ui/src/app/views/concepts/component.tsx

```diff
@@ -0,0 +1,420 @@
+      if (e.keyCode === 13) {
+        const searchTermLength = e.target.value.trim().length;
+        if (searchTermLength < 3) {
+          this.setState({showSearchError: true});
+        } else {
+          this.setState({currentSearchString: e.target.value}, this.searchConcepts);
+        }
+      }
+    }
+
+    selectDomain(domainCount: DomainCount) {
+      this.setState({selectedDomain: domainCount},
+        this.setConceptsAndVocabularies);
+    }
+
+    setConceptsAndVocabularies() {
+      const cacheItem = this.state.conceptsCache
+        .find(conceptDomain => conceptDomain.domain === this.state.selectedDomain.domain);
+      this.setState({concepts: cacheItem.items});
+      this.setState({
+        vocabularies: cacheItem.vocabularyList.map((vocabulary) => {
+          return {
+            ...vocabulary,
+            selected: true
+          };
+        })
+      });
+    }
+
+    async searchConcepts() {
+      const {standardConceptsOnly, currentSearchString, conceptsCache,
+        selectedDomain, completedDomainSearches} = this.state;
+      const {namespace, id} = this.props.workspace;
+      this.setState({concepts: [], searchLoading: true, searching: true, conceptsToAdd: [],
+        selectedConceptDomainMap: new Map<string, number>()});
+      const standardConceptFilter = standardConceptsOnly ?
+        StandardConceptFilter.STANDARDCONCEPTS : StandardConceptFilter.ALLCONCEPTS;
+
+      conceptsCache.forEach(async(conceptDomain) => {
+        const activeTabSearch = conceptDomain.domain === selectedDomain.domain;
+        const resp = await conceptsApi().searchConcepts(namespace, id, {
+          query: currentSearchString,
+          standardConceptFilter: standardConceptFilter,
+          domain: conceptDomain.domain,
+          includeDomainCounts: activeTabSearch,
+          includeVocabularyCounts: true,
+          maxResults: this.MAX_CONCEPT_FETCH
+        });
+        completedDomainSearches.push(conceptDomain.domain);
+        conceptDomain.items = this.convertToConceptInfo(resp.items);
+        conceptDomain.vocabularyList = resp.vocabularyCounts;
+        if (activeTabSearch) {
+          this.setState({
+            searchLoading: false,
+            conceptDomainCounts: resp.domainCounts,
+            selectedDomain: resp.domainCounts
+              .find(domainCount => domainCount.domain === conceptDomain.domain)});
+          this.setConceptsAndVocabularies();
+        }
+      });
+    }
+
+    selectConcept(concepts: ConceptInfo[]) {
+      const {selectedDomain, selectedConceptDomainMap} = this.state;
+      selectedConceptDomainMap[selectedDomain.domain] = concepts.filter(concept => {
+        return concept.domainId.toLowerCase() === selectedDomain.domain.toString().toLowerCase();
+      }).length;
+      this.setState({selectedConceptDomainMap: selectedConceptDomainMap, conceptsToAdd: concepts});
+    }
+
+    convertToConceptInfo(concepts: Concept[]): ConceptInfo[] {
```

Comment:
I actually think that the `ConceptInfo` object is now redundant, because it is basically just a `Concept` object with an attached field for whether or not the concept is selected, however with the react implementation we no longer need to store that piece of state there.  So I think I can get rid of both of those methods (`filterConceptSelection and `convertToConceptInfo`)